pyomo/contrib/pynumero/algorithms/solvers/scipy_solvers.py

```diff
@@ -124,6 +126,141 @@ def solve(self, x0=None):
         return results
 
 
+class _ScalarDenseSquareNlpSolver(DenseSquareNlpSolver):
+    # A base class for solvers for scalar equations.
+    # Not intended to be instantiated directly. Instead,
+    # NewtonNlpSolver or SecantNewtonNlpSolver should be used.
+
+    def __init__(self, nlp, timer=None, options=None):
+        super().__init__(nlp, timer=timer, options=options)
+        if nlp.n_primals() != 1:
+            raise RuntimeError(
+                "Cannot use the scipy.optimize.newton solver on an NLP with"
+                " more than one variable and equality constraint. Got %s"
+                " primals. Please use RootNlpSolver or FsolveNlpSolver instead."
+            )
+
+
+class NewtonNlpSolver(_ScalarDenseSquareNlpSolver):
+    """A wrapper around the SciPy scalar Newton solver for NLP objects
+
+    """
+
+    OPTIONS = ConfigBlock(
+        description="Options for SciPy newton wrapper",
+    )
```

Comment:
Instead of "overwriting" the class `OPTIONS` definition, You might consider "deriving" this OPTIONS from the base classes with either:
```python
OPTIONS = DenseSquareSolver.OPTIONS(
    description="Options for SciPy newton wrapper",
)
```
or
```python
OPTIONS = ConfifDict(
    description="Options for SciPy newton wrapper",
)
OPTIONS.declare_from(DenseSquareSolver.OPTIONS)
```

(The latter form accepts a `skip=` argument to omit certain entries from the source -- e.g., if you want to hide them or redeclare them with different default values.

Reply:
Done